### PR TITLE
ci: created API Explorer workflow

### PR DESCRIPTION
**Pull request by jkaster**

### CODE REVIEW

**Comment by bryans99** [comment]
When does this run?

**Review by bryans99 — DISMISSED** [review]
LGTM

File: .github/workflows/apix-ci.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+name: API Explorer CI
+on:
+  pull_request:
+    paths:
+      - packages/code-editor/**
+      - packages/run-it/**
+      - packages/api-explorer/**
+      - packages/extension-api-explorer/**
```
[the remark this answers is not in the document]
Comment (joeldodge79) [reply .github/workflows/apix-ci.yml:8]: we should definitely get a CI workflow file going for `extension-sdk*` paths

File: .github/workflows/apix-ci.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+env:
+  TS_JUNIT_OUTPUT_DIR: results/apix
+
+jobs:
+  unit:
+    name: Unit - ${{ matrix.os }} / Node ${{ matrix.node-version }}
+    env:
+      JEST_JUNIT_OUTPUT_DIR: results/apix
+      JEST_JUNIT_OUTPUT_NAME: ${{ matrix.os }}.tssdkrtl${{ matrix.node-version }}.xml
```
Comment (joeldodge79) [inline .github/workflows/apix-ci.yml:29]: cosmetic only:
```suggestion
      JEST_JUNIT_OUTPUT_NAME: ${{ matrix.os }}.apix${{ matrix.node-version }}.xml
```

File: .github/workflows/apix-ci.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    steps:
+      - run: exit 0
+
+  noop-python-results:
+    name: Python Tests
+    needs: [publish-test-results]
+    if: success() || failure()
+    runs-on: ubuntu-latest
+    steps:
+      - run: exit 0
```
Comment (joeldodge79) [inline .github/workflows/apix-ci.yml:109]:
```suggestion
      - run: exit 0

  noop-typescript-results:
    name: Typescript Tests
    needs: [publish-test-results]
    if: success() || failure()
    runs-on: ubuntu-latest
    steps:
      - run: exit 0
```

These noop tests are annoying but our best option to have ["required checks"](https://github.community/t/github-actions-and-required-checks-in-a-monorepo-using-paths-to-limit-execution/16586/2)  (which is still actually a work in progress)

Which means we also need the following block added to the end of these files:
* .github/workflows/codegen-ci.yml
* .github/workflows/python-ci.yml
* .github/workflows/tssdk-ci.yml

```yaml
  noop-typescript-results:
    name: APIX Tests
    needs: [publish-test-results]
    if: success() || failure()
    runs-on: ubuntu-latest
    steps:
      - run: exit 0
```

File: .github/workflows/apix-ci.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1.12
+        with:
+          check_name: Typescript Tests
```
Comment (joeldodge79) [inline .github/workflows/apix-ci.yml:82]:
```suggestion
          check_name: APIX Tests
```

**Review by joeldodge79 — CHANGES_REQUESTED** [review]
looking good, a few minor tweaks and the annoying `noop` stuff :-(

File: .github/workflows/apix-ci.yml (hunk 3 above)
Comment (joeldodge79) [reply .github/workflows/apix-ci.yml:109]: ug, scratch all this - gonna go with a different approach to "required checks"

Comment (joeldodge79) [inline .github/workflows/apix-ci.yml:109]:
```suggestion
```

let's get rid of the `noop` stuff here, gonna centralize it in a separate workflow file that triggers on every PR

**Review by joeldodge79 — CHANGES_REQUESTED** [review]
changed my mind on the `noop` stuff - let's just remove it. Feel free to delete all the `noop` jobs from the other workflow files or I can submit a separate PR for that - up to you

File: .github/workflows/apix-ci.yml (hunk 2 above)
Comment (jkaster) [reply .github/workflows/apix-ci.yml:29]: Good catch

**Review by joeldodge79 — APPROVED** [review]
👍

**Pull request merged**